app/services/idv/analytics_events_enhancer.rb

```diff
@@ -0,0 +1,62 @@
+module Idv
+  module AnalyticsEventsEnhancer
+    DECORATED_METHODS = %i[
+      idv_cancellation_confirmed
+      idv_cancellation_go_back
+      idv_cancellation_visited
+      idv_come_back_later_visit
+      idv_forgot_password
+      idv_forgot_password_confirmed
+      idv_final
+      idv_gpo_address_letter_enqueued
+      idv_gpo_address_letter_requested
+      idv_in_person_ready_to_verify_visit
+      idv_personal_key_acknowledgment_toggled
+      idv_personal_key_downloaded
+      idv_personal_key_submitted
+      idv_personal_key_visited
+      idv_phone_confirmation_form_submitted
+      idv_phone_confirmation_otp_rate_limit_attempts
+      idv_phone_confirmation_otp_rate_limit_locked_out
+      idv_phone_confirmation_otp_rate_limit_sends
+      idv_phone_confirmation_otp_resent
+      idv_phone_confirmation_otp_sent
+      idv_phone_confirmation_otp_submitted
+      idv_phone_confirmation_otp_visit
+      idv_phone_confirmation_vendor_submitted
+      idv_phone_error_visited
+      idv_phone_of_record_visited
+      idv_phone_otp_delivery_selection_visit
+      idv_phone_otp_delivery_selection_submitted
+      idv_proofing_resolution_result_missing
+      idv_review_complete
+      idv_review_info_visited
+      idv_setup_errors_visited
+      idv_start_over
+    ].freeze
+
+    extend ActiveSupport::Concern
+
+    included do
+      DECORATED_METHODS.each do |method_name|
+        alias_method "original_#{method_name}", method_name
+        define_method(method_name) do |**kwargs|
+          send("original_#{method_name}", **kwargs, **common_analytics_attributes)
+        end
+      end
+    end
```

Comment:
unsure if it would too "sneaky" but what if we used `prepend`?

so
```
class Analytics
  include AnalyticsEvents
  prepend Idv::AnalyticsEventsEnhancer
```

and then this becomes...
```suggestion
    DECORATED_METHODS.each do |method_name|
      define_method(method_name) do |**kwargs|
        super(**kwargs, **common_analytics_attributes)
      end
    end
```

Reply:
Hm, the specs for the enhancer class don't seem to like the change.

```diff
diff --git a/app/services/analytics.rb b/app/services/analytics.rb
index a70bdaddc..fee5bf485 100644
--- a/app/services/analytics.rb
+++ b/app/services/analytics.rb
@@ -2,7 +2,7 @@
 
 class Analytics
   include AnalyticsEvents
-  include Idv::AnalyticsEventsEnhancer
+  prepend Idv::AnalyticsEventsEnhancer
 
   attr_reader :user, :request, :sp, :ahoy
 
diff --git a/app/services/idv/analytics_events_enhancer.rb b/app/services/idv/analytics_events_enhancer.rb
index cd0460db1..398d8a7f1 100644
--- a/app/services/idv/analytics_events_enhancer.rb
+++ b/app/services/idv/analytics_events_enhancer.rb
@@ -35,14 +35,10 @@ module Idv
       idv_start_over
     ].freeze
 
-    extend ActiveSupport::Concern
-
-    included do
-      DECORATED_METHODS.each do |method_name|
-        alias_method "original_#{method_name}", method_name
-        define_method(method_name) do |**kwargs|
-          send("original_#{method_name}", **kwargs, **common_analytics_attributes)
-        end
+    DECORATED_METHODS.each do |method_name|
+      alias_method "original_#{method_name}", method_name
+      define_method(method_name) do |**kwargs|
+        send("original_#{method_name}", **kwargs, **common_analytics_attributes)
       end
     end
 
diff --git a/spec/services/idv/analytics_events_enhancer_spec.rb b/spec/services/idv/analytics_events_enhancer_spec.rb
index a2f6fe79e..f4ce17b0a 100644
--- a/spec/services/idv/analytics_events_enhancer_spec.rb
+++ b/spec/services/idv/analytics_events_enhancer_spec.rb
@@ -8,7 +8,7 @@ describe Idv::AnalyticsEventsEnhancer do
       @called_kwargs = kwargs
     end
 
-    include Idv::AnalyticsEventsEnhancer
+    prepend Idv::AnalyticsEventsEnhancer
 
     attr_reader :user, :called_kwargs
 ```

Failure:

```
$ rspec spec/services/idv/analytics_events_enhancer_spec.rb

An error occurred while loading ./spec/services/idv/analytics_events_enhancer_spec.rb.
Failure/Error: alias_method "original_#{method_name}", method_name

NameError:
  undefined method `idv_cancellation_confirmed' for module `Idv::AnalyticsEventsEnhancer'
```

Reply:
I'm happy to revisit this, but I won't consider it blocking since I'd like to get this merged today.